main.tf

```diff
@@ -64,123 +41,70 @@ resource "aws_security_group" "this_name_prefix" {
 ###################################
 # Security group rules with "cidr_blocks" and it uses list of rules names
 resource "aws_security_group_rule" "ingress_rules" {
-  count = local.create ? length(var.ingress_rules) : 0
-
-  security_group_id = local.this_sg_id
-  type              = "ingress"
-
-  cidr_blocks      = var.ingress_cidr_blocks
-  ipv6_cidr_blocks = var.ingress_ipv6_cidr_blocks
-  prefix_list_ids  = var.ingress_prefix_list_ids
-  description      = var.rules[var.ingress_rules[count.index]][3]
-
-  from_port = var.rules[var.ingress_rules[count.index]][0]
-  to_port   = var.rules[var.ingress_rules[count.index]][1]
-  protocol  = var.rules[var.ingress_rules[count.index]][2]
-}
-
-# Computed - Security group rules with "cidr_blocks" and it uses list of rules names
-resource "aws_security_group_rule" "computed_ingress_rules" {
-  count = local.create ? var.number_of_computed_ingress_rules : 0
+  for_each = local.create ? toset(var.ingress_rules) : toset([])
 
   security_group_id = local.this_sg_id
   type              = "ingress"
 
   cidr_blocks      = var.ingress_cidr_blocks
   ipv6_cidr_blocks = var.ingress_ipv6_cidr_blocks
   prefix_list_ids  = var.ingress_prefix_list_ids
-  description      = var.rules[var.computed_ingress_rules[count.index]][3]
+  description      = var.rules[each.value][3]
 
-  from_port = var.rules[var.computed_ingress_rules[count.index]][0]
-  to_port   = var.rules[var.computed_ingress_rules[count.index]][1]
-  protocol  = var.rules[var.computed_ingress_rules[count.index]][2]
+  from_port = var.rules[each.value][0]
+  to_port   = var.rules[each.value][1]
+  protocol  = var.rules[each.value][2]
 }
 
 ##########################
 # Ingress - Maps of rules
 ##########################
 # Security group rules with "source_security_group_id", but without "cidr_blocks" and "self"
 resource "aws_security_group_rule" "ingress_with_source_security_group_id" {
```

Comment:
all of these can go away to one generic ingress rule resource and one generic egress rule resource